process/dataValidators/txValidator.go

```diff
@@ -28,52 +30,54 @@ func NewTxValidator(accounts state.AccountsAdapter, shardCoordinator sharding.Co
 	return &TxValidator{
 		accounts:         accounts,
 		shardCoordinator: shardCoordinator,
-		rejectedTx:       uint64(0),
+		rejectedTxs:      uint64(0),
 	}, nil
 }
 
-// IsTxValidForProcessing is a nil implementation that will return true
+// IsTxValidForProcessing will filter transactions that needs to be added in pools
 func (tv *TxValidator) IsTxValidForProcessing(interceptedTx process.TxValidatorHandler) bool {
 	shardId := tv.shardCoordinator.SelfId()
-	txShardId := interceptedTx.GetSenderShardId()
-	if shardId != txShardId {
+	txShardId := interceptedTx.SenderShardId()
+	senderIsInAnotherShard := shardId != txShardId
+	if senderIsInAnotherShard {
 		return true
 	}
 
 	sndAddr := interceptedTx.GetSenderAddress()
 	accountHandler, err := tv.accounts.GetExistingAccount(sndAddr)
 	if err != nil {
-		log.Debug(fmt.Sprintf("Transaction sender address %s does not exits in current shard %d", sndAddr, shardId))
+		log.Debug(fmt.Sprintf("Transaction's sender address %s does not exist in current shard %d", sndAddr, shardId))
+		tv.rejectedTxs++
 		return false
 	}
 
 	accountNonce := accountHandler.GetNonce()
 	txNonce := interceptedTx.GetNonce()
-	if txNonce < accountNonce {
-		tv.rejectedTx++
+	lowerNonceInTx := txNonce < accountNonce
+	if lowerNonceInTx {
+		tv.rejectedTxs++
 		return false
 	}
 
 	account, ok := accountHandler.(*state.Account)
 	if !ok {
-		log.Error("Cannot convert account handler in a state.Account")
+		log.Error(fmt.Sprintf("Cannot convert account handler in a state.Account %v", sndAddr))
```

Comment:
might change the printing from %v to %s and then print it as hex

Reply:
Done.